src/extensions/replay/sessionrecording.ts

```diff
@@ -491,6 +507,12 @@ export class SessionRecording {
             }
         }
 
+        if (!_isNull(this._recordCanvas) && !_isNull(this._canvasFps) && !_isNull(this._canvasQuality)) {
```

Comment:
can we have a default fallback for `_canvasFps` and `_canvasQuality` instead of not enabling canvas at all?

Reply:
I considered that but didn't want to choose anything arbitrary before we knew what a sensible value might be. In reality they should never be null because we'll set the default on the backend so this is really more of a typing constraint

Reply:
the downside is that canvas recording can only start after the decide API has responded but by definition, recording should start recording until the decided responds, and only continue recording if it's enabled otherwise discard the recordings, so I guess we need to have some default values.

Reply:
IIRC what I meant above is related to cost control, if we don't need this right now, happy to merge as it is, approving it anyway.

Reply:
yep, better to start default disabled now because we can switch to default enabled easier than the reverse IMO

Reply:
Its not about the `_recordCanvas` but rather the `_canvasFps` nd `_canvasQuality`.

Based on that https://github.com/PostHog/product-internal/blob/main/requests-for-comments/2023-11-02-mobile-replay-specification.md

> before the decide response has been returned the client buffers all recording data

Without default (hardcoded) config for _canvasFps and _canvasQuality, this won't be possible since the 2 properties only have value after decide has responded.